examples/with-react-intl/components/PageWithIntl.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react'
+import {IntlProvider, addLocaleData, injectIntl} from 'react-intl'
+
+// Register React Intl's locale data for the user's locale in the browser. This
+// locale data was added to the page by `pages/_document.js`. This only happens
+// once, on initial page load in the browser.
+if (typeof window !== 'undefined' && window.ReactIntlLocaleData) {
```

Comment:
Thanks, I'm not familiar with CSP, will try to read more on this.

For detecting the browser locale I used `express-request-language`, then store it in the session, and allow the user to customise it (server-side).

For co-locating the code a bit more, I just manage to load the message files from within the HOC (it does involve a slight webpack tweak): https://github.com/relatenow/relate/blob/master/hocs/withIntl.js#L25